src/Compilers/CSharp/Portable/FlowAnalysis/DefinitelyAssignedWalker.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#if DEBUG
+// See comment in DefiniteAssignment.
+#define REFERENCE_STATE
+#endif
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// A region analysis walker that computes the set of variables that are definitely assigned
+    /// when a region is entered or exited.
+    /// </summary>
+    internal class DefinitelyAssignedWalker : AbstractRegionDataFlowPass
+    {
+        private readonly HashSet<Symbol> _definitelyAssignedOnEntry = new HashSet<Symbol>();
+        private readonly HashSet<Symbol> _definitelyAssignedOnExit = new HashSet<Symbol>();
+
+        private DefinitelyAssignedWalker(
+            CSharpCompilation compilation,
+            Symbol member,
+            BoundNode node,
+            BoundNode firstInRegion,
+            BoundNode lastInRegion)
+            : base(compilation, member, node, firstInRegion, lastInRegion)
+        {
+        }
+
+        internal static (HashSet<Symbol> entry, HashSet<Symbol> exit) Analyze(
+            CSharpCompilation compilation, Symbol member, BoundNode node, BoundNode firstInRegion, BoundNode lastInRegion)
+        {
+            var walker = new DefinitelyAssignedWalker(compilation, member, node, firstInRegion, lastInRegion);
+            try
+            {
+                bool badRegion = false;
+                walker.Analyze(ref badRegion, diagnostics: null);
+                return badRegion
+                    ? (new HashSet<Symbol>(), new HashSet<Symbol>())
+                    : (walker._definitelyAssignedOnEntry, walker._definitelyAssignedOnExit);
+            }
+            finally
+            {
+                walker.Free();
+            }
+        }
+
+        protected override void EnterRegion()
+        {
+            ProcessRegion(_definitelyAssignedOnEntry);
+            base.EnterRegion();
+        }
+
+        protected override void LeaveRegion()
+        {
+            ProcessRegion(_definitelyAssignedOnExit);
+            base.LeaveRegion();
+        }
+
+        private void ProcessRegion(HashSet<Symbol> definitelyAssigned)
+        {
+            // this can happen multiple times as flow analysis is multi-pass.  Always 
+            // take the latest data and use that to determine our result.
+            definitelyAssigned.Clear();
```

Comment:
This looks a lot like the AlwaysAssigned pass. Is there a way we could calculate both in a single pass?

Reply:
Great question!  I'll be 100% frank here.  I spent many hours reading through as much flow analysis code as i could, and I still find it pretty difficult to wrap my head around.  One of the things i *do* at least like and appreciate about the current API (both before and after my change) is that it at least tries to separate out the code passes to help keep complexity isolated.  

I can try to find a way to merge these.  However, personally, i really love how simple and easy it is to understand and maintain the code here.  It's more up to you on how to proceed.  Given that this stuff is likely at your fingertips, i imagine it's trivial in your head to have a pass that could do all this stuff while remaining clear to you.  But for me, it's not at that point :)

Reply:
I think it's fine to go forward with this as a separate pass, as it won't hurt compilation times. You should just be aware that flow analysis can be expensive and that this could increase IDE lag. I'll leave it up to the IDE perf team to make sure they're tracking this.

Reply:
Great.  Thanks.